### ISSUE TITLE
[TextInputLayout] Hint text overlaps outline

### ISSUE THREAD

**Issue by ivan200** [material-components/material-components-android#2811]
**Description:** outlinedbox crossed hint on android 4
[screenshot](https://github.com/ivan200/Overlap/blob/master/screen/Screenshot_20220712_010548.png?raw=true)

**Expected behavior:** Hint is displayed without overlapping with the box.

**Source code:** https://github.com/ivan200/Overlap

just simple app with Theme.Material3.Light theme 

**Minimal sample app repro:** 
Create new emulator -> Nexus 4 -> x86 Images -> Api 16 (without google api), run app
(my emulator settings [here](https://github.com/ivan200/Overlap/blob/master/avdSettings/config.ini) )

**Android API version:** 16

**Material Library version:** com.google.android.material:material:1.6.1

**Device:** Emulator api 16

ps also, TextInputLayout somehow ignores my set margins?

**Comment by drchen** [comment]
Does this happen on a real device (with API 16) as well?

**Comment by ivan200** [comment]
sorry i have only device with android 4.3 (api 18) and hint there fine
emulator with api 18 is fine also
but in emulator api 16 text is verlaps it is broken.

in manifest of this lib
android:minSdkVersion="14"

so think bug is still here.

**Comment by ivan200** [comment]
Yes, it is happened on devices also (Android 4.0.3 here)
[Photo](https://github.com/ivan200/Overlap/blob/master/screen/Photo_htc_api_15_20220730_145120.jpg?raw=true)

**Comment by drchen** [comment]
Great thank you so much for helping spot the issue! : )

Will try to find some time to work on this.

**Comment by henrik-lindqvist** [comment]
Any progress on this? The bug is still present in 1.8.0-alpha01.

**Comment by drchen** [comment]
hey sorry I'm pretty occupied these days. Will try to find some time to fix this before 1.8 is released.

**Comment by henrik-lindqvist** [comment]
@drchen Can the issue be worked around somehow, without patching the source code?

**Comment by drchen** [comment]
I can't think of a good way to do that. It's deep into the background drawable drawing logic....

**Comment by henrik-lindqvist** [comment]
@drchen Feared as much, guess i can't upgrade to higher than 1.4.0, as it seems the bug snuck in way back in 1.5.0.

**Comment by drchen** [comment]
yeah I'm sorry for that we didn't notice the issue on lower APIs sooner. It's somehow not uncovered by our screenshot tests. : (
